backend/ibutsu_server/widgets/compare_runs_view.py

```diff
@@ -0,0 +1,189 @@
+        apply_filters(Run.query, filters, Run)
+        .order_by(desc("start_time"))
+        .limit(run_limit or JJV_RUN_LIMIT)
+        .subquery()
+    )
+
+    # create the base query
+    query = (
+        session.query(
+            job_name.label("job_name"),
+            build_number.label("build_number"),
+            func.min(build_url.cast(Text)).label("build_url"),
+            func.min(env).label("env"),
+            func.min(Run.source).label("source"),
+            func.sum(Run.summary["xfailures"].cast(Integer)).label("xfailures"),
+            func.sum(Run.summary["xpasses"].cast(Integer)).label("xpasses"),
+            func.sum(Run.summary["failures"].cast(Integer)).label("failures"),
+            func.sum(Run.summary["errors"].cast(Integer)).label("errors"),
+            func.sum(Run.summary["skips"].cast(Integer)).label("skips"),
+            func.sum(Run.summary["tests"].cast(Integer)).label("tests"),
+            func.min(Run.start_time).label("min_start_time"),
+            func.max(Run.start_time).label("max_start_time"),
+            func.sum(Run.duration).label("total_execution_time"),
+            func.max(Run.duration).label("max_duration"),
+        )
+        .select_entity_from(sub_query)
+        .group_by(job_name, build_number)
+        .order_by(desc("max_start_time"))
+    )
+
+    # apply filters to the query
+    query = apply_filters(query, filters, Run)
+
+    # apply pagination and get data
+    query_data = query.offset(offset).limit(page_size).all()
+
+    # parse the data for the frontend
+    data = {
+        "jobs": [],
+        "pagination": {
+            "page": page,
+            "pageSize": page_size,
+            "totalItems": query.count(),  # TODO: examine performance here
+        },
+    }
+    for datum in query_data:
+        data["jobs"].append(
+            {
+                "_id": f"{datum.job_name}-{datum.build_number}",
+                "build_number": datum.build_number,
+                "build_url": datum.build_url,
+                "duration": (datum.max_start_time.timestamp() - datum.min_start_time.timestamp())
+                + datum.max_duration,  # noqa
+                "env": datum.env,
+                "job_name": datum.job_name,
+                "source": datum.source,
+                "start_time": datum.min_start_time,
+                "summary": {
+                    "xfailures": datum.xfailures,
+                    "xpasses": datum.xpasses,
+                    "errors": datum.errors,
+                    "failures": datum.failures,
+                    "skips": datum.skips,
+                    "tests": datum.tests,
+                    "passes": datum.tests - (datum.errors + datum.failures + datum.skips),
+                },
+                "total_execution_time": datum.total_execution_time,
+            }
+        )
+
+    return data
+
+
+def get_compare_runs_view(filter_=None, project=None, page=1, page_size=25, run_limit=None):
+    print('hello')
```

Comment:
hey there!

Reply:
You know how the initial debugging goes lol